### DISCUSSION TITLE
Is there any way to use `np.meshgrid(x,y)` in ndarray ?

### DISCUSSION THREAD

**Discussion by mdrokz** [rust-ndarray/ndarray#1231] [Q&A] [answered]
Im converting an ML model to rust and im translating some of the methods that use numpy and one method uses this `np.meshgrid` https://github.com/SakuraRiven/EAST/blob/cec7ae98f9c21a475b935f74f4c3969f3a989bd4/dataset.py#L232-L253 and i couldnt find this in ndarray

**Comment by zgbkdlm** [comment, marked as answer]
You need to use `itertools.cartesian_product` in https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.cartesian_product. 

Then convert the iterator to the ndarray by the method `from_iter`.

**Comment by jreniel** [comment]
Suggesting https://github.com/jreniel/meshgridrs as an alternative.

**Reply by bluss** [reply]
And linspace combined with broadcasting should work well.
